src/main/java/com/vivareal/search/api/adapter/FacetQueryAdapter.java

```
@@ -81,4 +85,16 @@ private void applyFacetsByNestedFields(
     if (nestedAgg.isPresent()) nestedAgg.get().subAggregation(agg);
     else searchRequestBuilder.addAggregation(nested(name, name).subAggregation(agg));
   }
+
+  private BucketOrder facetOrder(Item item) {
+    boolean isASC = item.getOrderOperator().equals(OrderOperator.ASC);
+
+    switch (item.getField().getName()) {
+      case "_key":
+        return key(isASC);
+      case "_count":
+      default:
+        return count(isASC);
```

I think that it is better to throw an `IllegalArgumentException` when the field is unknown. For instance, `facets=amenities _legalzinho` should result in a `400 Invalid Request` status response so that our clients are well aware that they are using an invalid parameter in the request. Failing to do so will "hide" errors when our clients have broken queries hitting us.

---

The parser will resolve this behavior. Please, take a look on **testInvalidFacetSortKey()** into **FacetParserTest.java**